NimbleTests/Matchers/PostNotificationTest.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  PostNotificationTest.swift
+//  Nimble
+//
+//  Created by Brian Gerstle on 12/28/15.
+//  Copyright © 2015 Jeff Hui. All rights reserved.
+//
+
+import XCTest
+import Nimble
+
+class PostNotificationTest: XCTestCase {
+// MARK: - Passing
+
+    func testPassesWhenNoNotificationsArePosted() {
+        expect(expression: postedNotifications {
+            // no notifications here!
+        }).to(beEmpty())
+    }
+
+    func testPassesWhenExpectedNotificationIsPosted() {
+        let testNotification = NSNotification(name: "Foo", object: nil)
+        expect(expression: postedNotifications {
+            NSNotificationCenter.defaultCenter().postNotification(testNotification)
+        }).to(equal([testNotification]))
+    }
+
+    func testPassesWhenAllExpectedNotificationsArePosted() {
+        let foo = NSObject()
+        let bar = NSObject()
+        let n1 = NSNotification(name: "Foo", object: foo)
+        let n2 = NSNotification(name: "Bar", object: bar)
+        expect(expression: postedNotifications {
+            NSNotificationCenter.defaultCenter().postNotification(n1)
+            NSNotificationCenter.defaultCenter().postNotification(n2)
+        }).to(equal([n1, n2]))
+    }
+
+// MARK: - Failing
+
+    func testFailsWhenNoNotificationsArePosted() {
+        let testNotification = NSNotification(name: "Foo", object: nil)
+        let pattern = try! "expected to equal <[NSConcreteNotification %addr% {name = Foo}]>, got <[]>"
```

Comment:
I think a the need for regular expression failure messages can be removed by using NSString's format to get the memory address. [beIdenticalTo's tests](https://github.com/Quick/Nimble/blob/master/NimbleTests/Matchers/BeIdenticalToTest.swift#L17) do this:

```swift
func testBeIdenticalToPositiveMessage() {
    let num1 = NSNumber(integer:1)
    let num2 = NSNumber(integer:2)
    let message = NSString(format: "expected to be identical to <%p>, got <%p>", num2, num1)
    failsWithErrorMessage(message.description) {
        expect(num1).to(beIdenticalTo(num2))
    }
}
```
I don't think there's a need to add regular expression matching for `failsWithErrorMessage`.